src/app/machines/views/MachineList/MachineList.test.tsx

```diff
@@ -705,4 +714,106 @@ describe("MachineList", () => {
     expect(fetches).toHaveLength(2);
     expect(fetches[fetches.length - 1].payload.params.page_number).toBe(2);
   });
+
+  it("shows a warning notification if not all controllers are configured with Vault", async () => {
+    const controllers = [
+      controllerFactory({
+        system_id: "abc123",
+        vault_configured: true,
+        node_type: NodeType.REGION_CONTROLLER,
+      }),
+      controllerFactory({
+        system_id: "def456",
+        vault_configured: false,
+        node_type: NodeType.REGION_AND_RACK_CONTROLLER,
+      }),
+    ];
+    state.controller.items = controllers;
+
+    renderWithBrowserRouter(
+      <MachineList searchFilter="" setSearchFilter={jest.fn()} />,
+      { wrapperProps: { state } }
+    );
+
+    expect(screen.getByTestId("vault-notification")).toHaveTextContent(
+      "Configure 1 other controller with Vault to complete this operation."
+    );
+  });
+
+  it("shows a warning notification if  all controllers are configured with Vault but secrets are not migrated", async () => {
+    const controllers = [
+      controllerFactory({
+        system_id: "abc123",
+        vault_configured: true,
+        node_type: NodeType.REGION_CONTROLLER,
+      }),
+      controllerFactory({
+        system_id: "def456",
+        vault_configured: true,
+        node_type: NodeType.REGION_AND_RACK_CONTROLLER,
+      }),
+    ];
+    state.controller.items = controllers;
+
+    renderWithBrowserRouter(
+      <MachineList searchFilter="" setSearchFilter={jest.fn()} />,
+      { wrapperProps: { state } }
+    );
+
+    expect(screen.getByTestId("vault-notification")).toHaveTextContent(
+      "Migrate your secrets to Vault to complete this operation."
+    );
+  });
+
+  it("doesn't show a warning notification if Vault setup has not been started", async () => {
+    const controllers = [
+      controllerFactory({
+        system_id: "abc123",
+        vault_configured: false,
+        node_type: NodeType.REGION_CONTROLLER,
+      }),
+      controllerFactory({
+        system_id: "def456",
+        vault_configured: false,
+        node_type: NodeType.REGION_AND_RACK_CONTROLLER,
+      }),
+    ];
+    state.controller.items = controllers;
+
+    renderWithBrowserRouter(
+      <MachineList searchFilter="" setSearchFilter={jest.fn()} />,
+      { wrapperProps: { state } }
+    );
+
+    expect(screen.queryByTestId("vault-notification")).not.toBeInTheDocument();
```

Comment:
Notifications have a `Incomplete Vault integration` title which can be used for querying instead of a test id.

Reply:
I tried that initially, but `getByTitle` doesn't seem to work for this, and `getByText` just gets the `h5` that contains the title as opposed to the whole div that the notification is contained within

I *could* use `.closest("div")`, but that returns a linting error: 
> Avoid direct Node access. Prefer using the methods from Testing Library.

Reply:
I don't see why you need to target the div that contains the notification, that's an implementation detail. Testing that the notification title is/is not on the screen is much better than using a test id. It's best to avoid them whenever you can. If you're interested in the _contents_ of the notification modal, you should test for the text itself.

I suggested using a title as it's displayed consistently for all types of notifications in the component you're testing it seems (just as the test id).